### ISSUE TITLE
Cannot pull image because of draconian rate limits by docker hub

### ISSUE THREAD

**Issue by ibotty** [varnish/docker-varnish#52]
Please consider publishing to github's container registry or quay.io.  Both have reasonable fair use policies at no cost.

```
> podman pull varnish
Trying to pull docker.io/library/varnish:latest...
WARN[0033] failed, retrying in 1s ... (1/3). Error: initializing source docker://varnish:latest: reading manifest latest in docker.io/library/varnish: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
[...!
```

**Comment by glensc** [comment]
auto-complete over the board or what is draconian?

**Comment by ibotty** [comment]
They have pretty bad rate limits for pulling images (or even checking for new tags) in place.  This is really bad when operating many computers behind a single IP.

**Comment by glensc** [comment]
you have totally ignored the question what is "draconian"?

**Comment by gquintard** [comment]
according to this [this](https://docs.docker.com/docker-hub/download-rate-limit/#what-is-the-download-rate-limit-on-docker-hub): 100 pulls per 6 hours

**Comment by glensc** [comment]
I don't know what's going on here. looks like they are talking about docker.io, but are blind or something can't see even draconian being a typo?

**Comment by gquintard** [comment]
@glensc , please tone down the aggressiveness a bit, everybody is trying to engage with you and you are being needlessly virulent in  return.

If you are asking about what "draconian" means, it's "excessively harsh and severe", and it fits the message here, so I do not believe it's a typo

**Comment by gquintard** [comment]
I finally had a look at this, and I'm sorry to say I'm closing it as WONTFIX.

The main reasoning is that it would increase the maintenance load with little gain and would create expectations we couldn't fulfill:
- users can already build images easily using this repository
- at the very least, pulling from the hub and pushing onto another registry is very easy to do at a user level
- just being logged in in docker hub as a free user increases the rate limit
- the docker hub constantly rebuild official images when the base images are updated to fix security issues, this isn't something we can commit to doing if using other registries

In short: working around the rate limit isn't a big deal for the user, and pushing to other registries would force us to take on a large burden that the hub currently does for us.

**Comment by gquintard** [comment]
@ibotty , @tianon mentioned on the hitch repository that the official images are mirrored on [ECR](https://gallery.ecr.aws/docker/library/varnish), hopefully this can be an alternative for you

**Comment by ibotty** [comment]
Yes, I saw that.  That resolves the issue.
